pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -115,27 +123,30 @@ func (p Provisioner) Provision(options *apibkt.BucketOptions) (*bktv1alpha1.Obje
 	if !bucketExists {
 		// if bucket already exists, this returns error: TooManyBuckets because we set the quota
 		// below. If it already exists, assume we are good to go
-		logger.Debugf("creating bucket %q", p.bucketName)
+		logger.Debugf("creating bucket %q owned by user %q", p.bucketName, p.cephUserName)
 		err = p.s3Agent.CreateBucket(p.bucketName)
 		if err != nil {
 			return nil, errors.Wrapf(err, "error creating bucket %q", p.bucketName)
 		}
-	} else if owner != options.UserID {
-		return nil, errors.Errorf("bucket %q already exists and is owned by %q for different OBC", p.bucketName, owner)
+	} else if owner != p.cephUserName {
+		logger.Debugf("bucket %q already exists and is owned by user %q instead of user %q, relinking...", p.bucketName, owner, p.cephUserName)
+
+		err = p.adminOpsClient.LinkBucket(p.clusterInfo.Context, admin.BucketLinkInput{Bucket: p.bucketName, UID: p.cephUserName})
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to link bucket %q to user %q", p.bucketName, p.cephUserName)
+		}
 	} else {
 		logger.Debugf("bucket %q already exists", p.bucketName)
 	}
 
-	singleBucketQuota := 1
-	_, err = p.adminOpsClient.ModifyUser(p.clusterInfo.Context, admin.User{ID: p.cephUserName, MaxBuckets: &singleBucketQuota})
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to set user %q bucket quota to %d", p.cephUserName, singleBucketQuota)
-	}
-	logger.Infof("set user %q bucket max to %d", p.cephUserName, singleBucketQuota)
-
-	additionalConfig, err := additionalConfigSpecFromMap(options.ObjectBucketClaim.Spec.AdditionalConfig)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to process additionalConfig")
+	// do not modify pre-existing users
+	if p.cephUserName == p.genUserName(options.ObjectBucketClaim) {
```

Comment:
The `genUserName()` function has changed over time. This will work for OBCs created since Sep 12, 2023, but older OBCs risk being modified.

Reply:
OK. We can probably abstract this check out as a a function that does multiple sanity checks.

Reply:
This should be addressed by ` isObcGeneratedUser()`